databuilder/databuilder/extractor/neo4j_extractor.py

```diff
@@ -43,7 +47,22 @@ def init(self, conf: ConfigTree) -> None:
         self.conf = conf.with_fallback(Neo4jExtractor.DEFAULT_CONFIG)
         self.graph_url = conf.get_string(Neo4jExtractor.GRAPH_URL_CONFIG_KEY)
         self.cypher_query = conf.get_string(Neo4jExtractor.CYPHER_QUERY_CONFIG_KEY)
-        self.driver = self._get_driver()
+
+        driver = conf.get(Neo4jExtractor.NEO4J_DRIVER, None)
+        if driver and isinstance(driver, Driver):
+            self.driver = driver
+        elif driver and not isinstance(driver, Driver):
```

Comment:
on the other hand, this is less important to check since usages of another type will result in "x does not have callable foo" type of errors